### DISCUSSION TITLE
요청 메시지를 파싱할 때, 웹 서버는 어떤 일을 하나요?

### DISCUSSION THREAD

**Discussion by sa02045** [prgrms-web-devcourse/We-are-the-MASTERS-OF-HTTP#36] [Interview Question]
웹 서버가 요청 메시지를 파싱하는 단계를 알려주세요

**Comment by Hyevvy** [comment]
웹 서버가 요청 메시지를 파싱하는 단계는 다음과 같습니다.
1. 요청줄 파싱하여 method, URI, HTTP version을 찾는다.
2. 메시지 헤더를 읽는다.
3. 헤더의 끝을 의미하는 CRLF를 찾는다.
4. 요청 본문을 읽는다.

**Comment by punchdrunkard** [comment]
요청 메시지는 첫 줄은 요청줄이 오고 이어서 여러 종류의 헤더가 나온 다음에 줄바꿈을 두번하고(CRLF) 메시지 바디가 오는 형식으로 이루어져 있습니다.
따라서 이 형식에 따라 

(이하의 내용은 혜경님과 같음...)
> 웹 서버가 요청 메시지를 파싱하는 단계는 다음과 같습니다.
> 
> 1. 요청줄 파싱하여 method, URI, HTTP version을 찾는다.
> 2. 메시지 헤더를 읽는다.
> 3. 헤더의 끝을 의미하는 CRLF를 찾는다.
> 4. 요청 본문을 읽는다.

이 때, 요청 본문을 읽는 경우 `Content-Length` 헤더를 이용하여 메시지를 올바르게 구분할 수있습니다.
